app/cdap/components/WidgetSVG/utils.tsx

```diff
@@ -0,0 +1,118 @@
+import DataPrepStore from 'components/DataPrep/store';
+import DataPrepActions from 'components/DataPrep/store/DataPrepActions';
+import WidgetSVG from 'components/WidgetSVG';
+import {
+  IConnectorTypesWithSVG,
+  IConnectorDetailsPayload,
+  IConnectorTypes,
+} from 'components/WidgetSVG/types';
+import { importDatasetIcon } from 'components/WrangleHome/Components/WrangleCard/iconStore/importDataset';
+import React from 'react';
+
+export const getWidgetData = async () => {
+  const connectorTypes: IConnectorTypesWithSVG[] = await fetchConnectors();
+  const connectorsTypesData: IConnectorTypes[] = [];
+  const IConnectionWithConnectorType: IConnectorTypesWithSVG[] = [];
+  const allConnectorsPluginProperties: Map<
+    string,
+    IConnectorDetailsPayload[]
+  > = getCategoriesToConnectorsMap(connectorTypes);
+  const connectionPayload: IConnectorDetailsPayload[] = [];
+  allConnectorsPluginProperties?.forEach((eachProperty) => {
+    if (eachProperty.length) {
+      eachProperty.forEach((item) => {
+        connectionPayload.push(item);
+      });
+    }
+  });
+
+  const connectionDetailsData = await Promise.all(
+    connectionPayload.map(async (item, index) => {
+      const selectedConnector = {
+        artifact: item.artifact,
+        category: item.category,
+        name: item.name,
+        type: item.type,
+      };
+      connectorsTypesData.push(selectedConnector);
+      return new Promise((resolve, reject) => {
+        const response = fetchConnectionDetails(selectedConnector);
+        if (response) {
+          resolve(response);
+        }
+      });
+    })
+  );
+  const connectorWidgetJson =
+    Array.isArray(connectionDetailsData) &&
+    connectionDetailsData.length &&
+    connectionDetailsData.map(({ connectorWidgetJSON }) => connectorWidgetJSON);
+
+  connectorsTypesData.map((connectorType) => {
+    let connectorTypeHasWidget: boolean = false;
+    /**
+     * Getting widget icons for connector types
+     */
+    Array.isArray(connectorWidgetJson) &&
+      connectorWidgetJson.length &&
+      connectorWidgetJson.map((eachConnector) => {
+        if (
+          eachConnector['display-name'] &&
+          eachConnector['display-name'].includes(connectorType.name)
+        ) {
+          IConnectionWithConnectorType.push({
+            ...connectorType,
+            SVG: (
+              <WidgetSVG
+                imageSource={eachConnector?.icon?.arguments?.data}
+                label={connectorType.name}
+              />
+            ),
+          });
+          connectorTypeHasWidget = true;
+        }
+      });
+    /**
+     * Retaining the connector types which are not part of widget api
+     */
+    if (!connectorTypeHasWidget) {
+      IConnectionWithConnectorType.push({
+        ...connectorType,
+        SVG: <WidgetSVG imageSource={undefined} label={connectorType.name} />,
+      });
+    }
+  });
+
+  IConnectionWithConnectorType.push({
+    name: 'Imported Dataset',
```

Comment:
we might be using this name string to display the text in the button
plz use i18n